setup.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+from setuptools import setup, find_packages
+
+# Get description from README.md
+LONG_DESCRIPTION = ''
+with open('README.md', encoding='utf-8') as rd:
+    LONG_DESCRIPTION = rd.read()
+
+# Here we cannot import the version but need to read it otherwise we might have an ImportError
+# during execution of the setup.py if the package contains other libaries.
+VERSION_LINE = list(filter(lambda l: l.startswith('VERSION'),
```

Comment:
What kind of import error are you getting / what is the cause for them ? They might happen if we have a loop within import with a 'from ** import ***', but this doesn't look like this would be the case here. Am I missing something completely ?

Reply:
It would raise ImportError if the file with the version aka __init__.py would import other stuff that are not python builtin (external library). setup.py script is usually not run within the virtual environment and can also be run from a python2 version. This means that external library cannot be imported in the setup.py script. And again as mentioned in PyPA reading the version by importing it is a bad practice.